prospero-cli/src/main/java/org/wildfly/prospero/cli/commands/UpdateCommand.java

```diff
@@ -434,6 +437,24 @@ private static boolean buildUpdate(UpdateAction updateAction, Path updateDirecto
         return true;
     }
 
+    private static boolean buildUpdate(UpdateAction updateAction, Path updateDirectory, boolean yes, CliConsole console, Supplier<Boolean> confirmation, Path installationDir) throws OperationException, ProvisioningException {
+        final UpdateSet updateSet = updateAction.findUpdates();
+
+        console.updatesFound(updateSet.getArtifactUpdates());
+        if (updateSet.isEmpty()) {
+            return false;
+        }
+
+        if (!yes && !confirmation.get()) {
+            return false;
+        }
+
+        updateAction.buildUpdate(updateDirectory.toAbsolutePath());
+        writeChannelMapToFile(updateSet, installationDir);
```

Comment:
Could you move this to `PrepareCandidateAction`? The updateSet is available in `UpdateAction.buildUpdate` and can be passed to the `PrepareCandidateAction`